src/app/DeviceSelector.tsx

```diff
@@ -91,12 +95,28 @@ const getBoardControllerVersion = async (
     dispatch(setBoardControllerFirmwareVersion(bcVersion.bc_fw_ver));
 };
 
+const getCurrentBoardControllerConfig = async (
+    dispatch: AppDispatch,
+    device: Device
+) => {
+    if (!device) {
+        return;
+    }
```

Comment:
How can the device here be undefined?

Reply:
Removed the check - I came to the same conclusion as you.